routes/quiz.js

```diff
@@ -10,70 +10,114 @@ var players_model = require('../models/players.js');
-                    players = players_model.fetchPlayers(type, doc.team_id, doc.rb_team_id, doc.league, doc.usat_id, res, players_model.intreturnPlayers, players_model.returnPlayers);
-                });
-            }
-            else{
-              players = players_model.fetchPlayers(type, team_id, rb_team_id, league, usat_id, res, req, players_model.intreturnPlayers, players_model.returnPlayers);
-            }
-          });
+        else{
+          players = players_model.fetchPlayers(type, team_id, rb_team_id, league, usat_id, res, req, players_model.intreturnPlayers, players_model.returnPlayers);
         }
+      });
+    }
+    // Calculate all other scores for this team in the background
+    fetchQuizScores(req, rb_team_id);
   });
-});
+})
  
+
 router.get('/results', function(req, res) {
   var quiz_id = req.query.quiz_id,
     quiz_score = req.query.quiz_score,
-    possible_score = req.query.possible_score;
+    possible_score = req.query.possible_score,
+    percentage_correct = req.query.percentage_correct;
     db.open(function(err, db){
       db.collection("quiz").update({_id: quiz_id},
-      {$set: {quiz_score: quiz_score, possible_score: possible_score}},
+      {$set: {quiz_score: quiz_score, possible_score: possible_score, percentage_correct: percentage_correct}},
       {upsert: true, multi:false}, function (err, upserted){
         if (err){
           console.log(err);
           res.json({success: false});
         }
         else {
-          res.json({success: true});
+          res.json({all_scores: req.session.scores, this_score: percentage_correct, success: true});
         }
       });
     });
 });
+
+
+// Pull all raw quiz scores for that team_id
+var fetchQuizScores = function(req, rb_team_id){
+  db.collection('quiz').find({ "rb_team_id" : rb_team_id}, {percentage_correct: 1}, function(err, items){
+    percentages = []
+    if (err){
+      console.log(err);
+    }
+    else {
+      for (i=0;i<Object.keys(items).length;i++){
+        percentages.push(items[i].percentage_correct);
+      }
+      // Assign each quiz score to a bucket for graph display
+      bracketQuizScores(req, percentages);
+    }
+  })
+}
+
+
+var bracketQuizScores = function(req, percentage_array){
+  aScores = 0;
+  bScores = 0;
+  cScores = 0;
+  dScores = 0;
+  eScores = 0;
+  
+  for (i=0;i<percentage_array.length;i++){
+    if (percentage_array[i] == null){console.log("meh")}
+    else if (percentage_array[i] < 0.20){aScores++}
+    else if (percentage_array[i]>= 0.20 && percentage_array[i] < 0.40){bScores++}
+    else if (percentage_array[i]>= 0.40 && percentage_array[i] < 0.60){cScores++}
+    else if (percentage_array[i]>= 0.60 && percentage_array[i] < 0.80){dScores++}
+    else if (percentage_array[i] >= 0.80){eScores++}
+    else{console.log("ignore")}
+  }
+  req.session.scores = {a: aScores, b:bScores , c:cScores, d:dScores, e:eScores};
```

Comment:
be nice to name these keys something more descriptive so that person doing FE knows exactly what to do with them.